pages/api/pull-request/index.ts

```diff
@@ -96,50 +102,132 @@ async function post(req: NextApiRequest, res: NextApiResponse) {
       issueId: issue.id,
       githubId: created.data?.number,
       githubLogin: username,
-      branch
+      branch,
+      status: "pending"
     });
 
-    issue.state = "ready";
-
-    const issueLink = `${process.env.NEXT_PUBLIC_HOME_URL}/bounty?id=${issue?.githubId}&repoId=${issue?.repository_id}`;
-    const body = `@${issue.creatorGithub}, @${username} has a solution - [check your bounty](${issueLink})`;
-    await octoKit.rest.issues.createComment({
-      owner,
-      repo,
-      issue_number: issue.githubId,
-      body
-    });
-
-    await issue.save();
     /*twitterTweet({
       type: 'bounty',
       action: 'solution',
       username: username,
       issue
     })*/
-    await api.post(`/seo/${issue?.issueId}`).catch((e) => {
-      console.log("Error creating SEO", e);
-    });
 
-    return res.json("ok");
+    return res.status(200).json({ 
+      bountyId: issue.contractId,
+      originRepo: repoInfo.githubPath,
+      originBranch: issue.branch,
+      originCID: issue.issueId,
+      userRepo: `${username}/${repo}`,
+      userBranch: branch,
+      cid: created.data?.number 
+    });
   } catch (error) {
     return res.status(error.response.status).json(error.response.data);
   }
 }
 
-export default async function PullRequest(req: NextApiRequest,
-                                          res: NextApiResponse) {
+async function del(req: NextApiRequest, res: NextApiResponse) {
+  const { 
+    repoId: repository_id, 
+    issueGithubId: githubId, 
+    bountyId: contractId, 
+    issueCid: issueId, 
+    pullRequestGithubId, 
+    customNetworkName,
+    creator,
+    userBranch,
+    userRepo
+  } = req.body;
+
+  const customNetwork = await models.network.findOne({
+    where: {
+      name: {
+        [Op.iLike]: String(customNetworkName)
+      }
+    }
+  });
+
+  if (!customNetwork || customNetwork?.isClosed) return res.status(404).json("Invalid");
+
+  const issue = await models.issue.findOne({
+    where: {
+      issueId,
+      network_id: customNetwork.id,
+      contractId,
+      repository_id,
+      githubId
+    },
+    include: [
+      { association: "repository" }
+    ]
+  });
+
+  if (!issue) return res.status(404).json("Invalid");
+
+  const pullRequest = await models.pullRequest.findOne({
+    where: {
+      githubId: String(pullRequestGithubId),
+      githubLogin: creator,
+      branch: userBranch,
+      status: "pending"
+    }
+  });
+
+  if (!pullRequest) return res.status(404).json("Invalid");
+
+  const network = networkBeproJs({
+    contractAddress: customNetwork.name.toLowerCase() === publicRuntimeConfig.networkConfig.networkName.toLowerCase() ? 
+      publicRuntimeConfig.contract.address : customNetwork.networkAddress,
+    version: 2
+  }) as Network_v2;
+
+  await network.start();
+
+  const networkBounty = await network.getBounty(contractId);
+  
+  if (!networkBounty) return res.status(404).json("Invalid");
+  
+  if (networkBounty.pullRequests.find(pr =>
```

Comment:
for sanity sake, assign the value on this condition to a variable and then if that variable (or simplify the if through better spacing, or make a function for the finder)